src/main/java/org/junit/internal/runners/statements/RunBefores.java

```diff
@@ -20,6 +31,9 @@ public RunBefores(Statement next, List<FrameworkMethod> befores, Object target)
 
     @Override
     public void evaluate() throws Throwable {
+        if (null != previous) {
```

Comment:
With my patch the class BlockJUnit4ClassRunner uses null as value for previous. By providing the old constructor, previous will be null, too.